### PR TITLE
enable timerotateloghandler (configurable)

### PR DESCRIPTION
**Pull request by sid88in**

### CODE REVIEW

**Comment by sid88in** [comment]
@mistercrunch I am testing this PR to enhance logging in caravel.  Reference: 

1) http://docs.python-guide.org/en/latest/writing/logging/
2) https://docs.python.org/2/library/logging.config.html

File: caravel/caravel_logging.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+Settings for root logger.
+1) Log messages will be printed to console a
+2) also to log file (rotated, with specified size).
+
+Reference:
+1) http://docs.python-guide.org/en/latest/writing/logging/
+2) https://docs.python.org/2/library/logging.config.html
+"""
+
+from caravel import app
```
Comment (mistercrunch) [inline caravel/caravel_logging.py]: I think that creates a circular import

**Comment by mistercrunch** [comment]
The build failed because of what I'm guessing is a circular import. 

I don't think we need a specific module for just the logging config, I'd squeeze that into `conf.py` instead. To avoid circular imports you can simply use a function that receives the app configuration and returns the loggingconfig object perhaps.

**Comment by sid88in** [comment]
Cool. Looking into it.

File: caravel/bin/caravel [hunk 2]
```diff
@@ -17,6 +17,8 @@ from caravel import data, utils
 from caravel import db
 
 config = app.config
+logging.config.dictConfig(config.get('LOGGING_CONFIG'))
```
Comment (coleslaw) [inline caravel/bin/caravel]: is this line needed here and everywhere else? 
It's already called in `caravel/__init__.py` and that is executed whenever any caravel module is imported.

@sid88in

Comment (sid88in) [inline caravel/bin/caravel]: Thats a good catch. I tested it after removing the redundant declaration and it works.

File: caravel/bin/caravel [hunk 3]
```diff
@@ -36,8 +37,9 @@ manager.add_command('db', MigrateCommand)
     help="Specify the timeout (seconds) for the gunicorn web server")
 def runserver(debug, port, timeout, workers):
     """Starts a Caravel web server"""
-    debug = debug or config.get("DEBUG")
-    if debug:
+    debug = config.get('LOG_LEVEL')
```
Comment (mistercrunch) [inline caravel/bin/caravel]: `debug` meant something else here (whether the web server should run in debug mode). We should keep that insulated from the logging level.

Comment (sid88in) [inline caravel/bin/caravel]: I see. I noticed that but was wondering what if debug = True and Log Level = Debug.  I have reverted back the setting for running web server as debug mode true/false

**Comment by mistercrunch** [comment]
Ok sorry about the incremental comments, I should have written all comments at once early on.

So I think the normal assumption is that a piece of software logs to STDOUT only so that if I run it with something like `runit` or `sv` it will do its own log rotation and all that. Can we make the default to only log to STDOUT and make it easy and clear for people to configure it to log differently?

**Comment by sid88in** [comment]
Sure. I'll look into making it configurable by passing boolean for log handers.

**Comment by sid88in** [comment]
@mistercrunch I have configured timeRotateLogRotation which is often needed when this application runs over large scale - logs can then be distributed across multiple files automatically based on rollover time (in this case every midnight new file will be created). Also I have reverted back all logger statements to print (in master), so people can configure how they want to log. TimeRotate is also configurable in conifg.

File: caravel/__init__.py [hunk 4]
```diff
@@ -30,6 +27,18 @@
 
 migrate = Migrate(app, db, directory=APP_DIR + "/migrations")
 
+# Logging configuration
+logging.basicConfig(format='%(asctime)s:%(levelname)s:%(name)s:%(message)s')
+logging.getLogger().setLevel(logging.DEBUG)
```
Comment (mistercrunch) [inline caravel/__init__.py]: if only two things was to get parameterized it should be the logging level and the format string

Comment (sid88in) [inline caravel/__init__.py]: fixed

**Comment by mistercrunch** [comment]
This is good to go, you just need to rebase...

**Pull request merged**